packages/wallets/src/safe/pages/transaction.page.ts

```diff
@@ -0,0 +1,199 @@
+import { Locator, Page, test, expect } from '@playwright/test';
+import { ConsoleLogger } from '@nestjs/common';
+import { WalletPage } from '../../wallet.page';
+import { WalletConnectTypes } from '../../wallets.constants';
+import { Big } from 'big.js';
+
+export class TransactionPage {
+  logger = new ConsoleLogger(`Safe. ${TransactionPage.name}`);
+  transactionCardContent: Locator;
+  contractExplorerUrl: Locator;
+  tokenAmount: Locator;
+  executeTxBtn: Locator;
+  finishTxBtn: Locator;
+  switchNetworkBtn: Locator;
+
+  allActionsList: Locator;
+  actionItem: Locator;
+  contentLoader: Locator;
+  transactionFailBanner: Locator;
+
+  constructor(
+    public page: Page,
+    public extensionPage: WalletPage<WalletConnectTypes.EOA>,
+  ) {
+    this.transactionCardContent = this.page.getByTestId('card-content').first();
+    this.contractExplorerUrl = this.transactionCardContent
+      .getByTestId('explorer-btn')
+      .first();
+    this.tokenAmount = this.transactionCardContent.getByTestId('token-amount');
+    this.executeTxBtn = this.page.getByTestId('execute-form-btn');
+    this.finishTxBtn = this.page.getByTestId('finish-transaction-btn');
+    this.switchNetworkBtn = this.page.getByText('Switch to');
+    this.allActionsList = this.page.getByTestId('all-actions');
+    this.actionItem = this.page.getByTestId('action-item');
+    this.contentLoader = this.page.locator('#gooey').locator('../../..');
+    this.transactionFailBanner = this.page.getByText(
+      'This transaction will most likely fail.',
+    );
+  }
+
+  async assertsContractOfTransaction(expectedAddress: string) {
+    await this.waitForLoaderToBeHidden();
```

Comment:
Could you wrap `this.waitForLoaderToBeHidden();` helper into corresponding functions since this helper where it's possible?